### PR TITLE
SALTO-1508 standard picklist fix

### PR DESCRIPTION
**Pull request by emaohi**
Fix  standard picklist change-validator that reports false-positive on errors.
Change includes referring to the correct value-set annotation (valueSetName instead of valueSet) and checking that the referred value set is indeed standard (standardValueSet).

---

_Release Notes_: 

Salesforce adapter:
_NA_

---

_User Notifications_: 
_NA_

### CODE REVIEW

File: packages/salesforce-adapter/src/filters/value_set.ts [hunk 1]
```diff
@@ -37,10 +37,10 @@ export const isPicklistField = (changedElement: ChangeDataType): changedElement
       Types.primitiveDataTypes.MultiselectPicklist.elemID.getFullName(),
     ]).includes(changedElement.refType.elemID.getFullName())
 
-export const isStandardValueSetPicklistField = (field: Field): boolean =>
-  field.annotations[FIELD_ANNOTATIONS.VALUE_SET] instanceof ReferenceExpression
+export const isValueSetReference = (field: Field): boolean =>
+  field.annotations[VALUE_SET_FIELDS.VALUE_SET_NAME] instanceof ReferenceExpression
```
Comment (netama) [inline packages/salesforce-adapter/src/filters/value_set.ts:41]: (doesn't seem related to this PR) it might just be a naming issue, but if `valueSetName` is the name of an annotation then we should get it from somewhere other than `VALUE_SET_FIELDS`. @ori-moisis / @noamhammer should we just change the const name?

(+ may be better to switch to `isReferenceExpression`, though it's also not a change from this PR)

File: packages/salesforce-adapter/src/filters/value_set.ts [hunk 2]
```diff
@@ -37,10 +37,10 @@ export const isPicklistField = (changedElement: ChangeDataType): changedElement
...
 
-export const isGlobalValueSetPicklistField = (field: Field): boolean =>
+export const hasValueSetNameAnnotation = (field: Field): boolean =>
```
Comment (netama) [inline packages/salesforce-adapter/src/filters/value_set.ts:43]: nit: if a "logical" function name is too long, maybe add a comment about when the `valueSetName` annotation is used (I think you mentioned it's referencing global/standard?)

File: packages/salesforce-adapter/src/change_validators/picklist_standard_field.ts [hunk 3]
```diff
@@ -18,13 +18,22 @@ import {
...
 import { apiName, isCustom } from '../transformers/transformer'
-import { isPicklistField, isStandardValueSetPicklistField } from '../filters/value_set'
+import { isPicklistField, isValueSetReference } from '../filters/value_set'
+import { STANDARD_VALUE_SET } from '../filters/standard_value_sets'
+import { isInstanceOfType } from '../filters/utils'
+import { VALUE_SET_FIELDS } from '../constants'
 
 const { awu } = collections.asynciterable
 
+const isStandardValueSet = async (plField: Field): Promise<boolean> => {
```
Comment (netama) [inline packages/salesforce-adapter/src/change_validators/picklist_standard_field.ts:28]: nit: what does `plField` stand for? in general there's a preference here for longer-and-meaningful names if the shorter ones are not clear :)

File: packages/salesforce-adapter/src/change_validators/picklist_standard_field.ts [hunk 4]
```diff
@@ -18,13 +18,22 @@ import {
...
+  const standardVSchecker = isInstanceOfType(STANDARD_VALUE_SET)
+  return isValueSetReference(plField)
+  && standardVSchecker(plField.annotations[VALUE_SET_FIELDS.VALUE_SET_NAME].value)
```
Comment (netama) [inline packages/salesforce-adapter/src/change_validators/picklist_standard_field.ts:31]: a few comments on this:
1. (nit) IMO this is a little hard to read (especially since we don't have semicolons at the end of lines) - please either indent the 2nd line or add parentheses
```suggestion
  return isValueSetReference(plField)
    && standardVSchecker(plField.annotations[VALUE_SET_FIELDS.VALUE_SET_NAME].value)
```
```suggestion
  return (
    isValueSetReference(plField)
    && standardVSchecker(plField.annotations[VALUE_SET_FIELDS.VALUE_SET_NAME].value)
  )
```
2. seems we're "losing" a chance to use safe extraction logic with `.annotations[VALUE_SET_FIELDS.VALUE_SET_NAME]` here - maybe add a `getValueSetReference` so that we know it exists? if not, we should allow for the chance that it becomes undefined (use `?.` instead of `.`)
3. I'm not sure we can assume that the reference's `value` exists at this point - I think you should use `getResolvedValue` instead

**Review by netama — COMMENTED** [review]
looks good :) added a few nits and one potential bug

File: packages/salesforce-adapter/src/filters/value_set.ts (hunk 1 above)
Comment (ori-moisis) [reply packages/salesforce-adapter/src/filters/value_set.ts:41]: seems like `VALUE_SET_FIELDS` contains the names of the annotations on picklist fields. maybe it should just be renamed to `PICKLIST_FIELD_ANNOTATIONS`?

File: packages/salesforce-adapter/src/change_validators/picklist_standard_field.ts (hunk 4 above)
Comment (ori-moisis) [reply packages/salesforce-adapter/src/change_validators/picklist_standard_field.ts:31]: @netama AFAIK currently everything in the adapter (including change validators) get fully resolved elements

File: packages/salesforce-adapter/src/filters/value_set.ts (hunk 1 above)
Comment (netama) [reply packages/salesforce-adapter/src/filters/value_set.ts:41]: :+1: that's what I was asking :)

File: packages/salesforce-adapter/src/change_validators/picklist_standard_field.ts (hunk 4 above)
Comment (netama) [reply packages/salesforce-adapter/src/change_validators/picklist_standard_field.ts:31]: got it, so ignore 3 :)

**Review by netama — APPROVED** [review]
LGTM (please look at the remaining comments before merging)

File: packages/salesforce-adapter/src/filters/value_set.ts (hunk 1 above)
Comment (emaohi) [reply packages/salesforce-adapter/src/filters/value_set.ts:41]: for now changed to `isReferenceExpression`

File: packages/salesforce-adapter/src/filters/value_set.ts (hunk 2 above)
Comment (emaohi) [reply packages/salesforce-adapter/src/filters/value_set.ts:43]: not really sure what you've meant here? the previous name meaning was wrong - it didn't check that the picklist had global value set (not global and not a field), but just a reference to a value set in its annotations.

File: packages/salesforce-adapter/src/change_validators/picklist_standard_field.ts (hunk 3 above)
Comment (emaohi) [reply packages/salesforce-adapter/src/change_validators/picklist_standard_field.ts:28]: picklist, changed :)

File: packages/salesforce-adapter/src/change_validators/picklist_standard_field.ts (hunk 4 above)
Comment (emaohi) [reply packages/salesforce-adapter/src/change_validators/picklist_standard_field.ts:31]: re 2 - the `isValueSetReference(plField)` check assueres that the field has this annotation, so i guess it's ok to just access it in the right part of the `&&`

File: packages/salesforce-adapter/src/filters/value_set.ts (hunk 2 above)
Comment (netama) [reply packages/salesforce-adapter/src/filters/value_set.ts:43]: understood :) I just meant that `hasValueSetNameAnnotation` is very technical and maybe we can clarify what this is really checking / when it would be truthy (e.g. truthy when the field is a picklist field that is referencing either a standard or a global value set, falsy when it's either referencing an in-place value set or when it's not a picklist). the case breakdown I wrote is probably not exact - I just think that `hasValueSetNameAnnotation(field)` doesn't add much on top of `field.annotations[VALUE_SET_FIELDS.VALUE_SET_NAME] !== undefined`

Comment (emaohi) [reply packages/salesforce-adapter/src/filters/value_set.ts:43]: removed the function and just added the check inline
